lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -872,8 +869,6 @@ Optional<MemOp::PortKind> MemOp::getPortKind(size_t resultNo) {
 
 /// Return the data-type field of the memory, the type of each element.
 FIRRTLType MemOp::getDataType() {
-  assert(getNumResults() != 0 && "Mems with no read/write ports are illegal");
-
```

Comment:
Pedantically, I don't think this is actually checked in the verifier for `MemOp`, it's just that we remove no-port memories during parsing.

It would be good to add a verification check in `verifyMemOp` (in `FIRRTLOps.cpp`) that errors out if the memory has no ports.

Reply:
Yep, that would be a good verification hook to add.  Please keep these asserts though - they cause no harm to the API and catch IR that is in an inconsistent state.